lib/cmd_tool_next.go

```diff
@@ -0,0 +1,127 @@
+package lib
+
+import (
+	"fmt"
+	"net"
+	"github.com/spf13/pflag"
+)
+
+type CmdToolNextFlags struct {
+	Help  bool
+	Quiet bool
+}
+
+func (f *CmdToolNextFlags) Init() {
+	pflag.BoolVarP(
+		&f.Help,
+		"help", "h", false,
+		"show help.",
+	)
+	pflag.BoolVarP(
+		&f.Quiet,
+		"quiet", "q", false,
+		"quiet mode; suppress additional output.",
+	)
+}
+
+func CmdToolNext(
+	f CmdToolNextFlags,
+	args []string,
+	printHelp func(),
+) error {
+	if f.Help {
+		printHelp()
+		return nil
+	}
+
+	increment := 1
+	actionFunc := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_IP:
+			newIP := ipAdd(input, increment)
+			if newIP != nil {
+				fmt.Println(newIP)
+			}
+		default:
+			return ErrNotIP
+		}
+		return nil
+	}
+	err := GetInputFrom(args, true, true, actionFunc)
+	if err != nil {
+		fmt.Println(err)
+	}
+	
+	return nil
+}
+
+func ipAdd(input string, delta int) net.IP {
+	ip := net.ParseIP(input)
+	if ip != nil {
+		if ip.To4() != nil {
+			ipInt := ipToUint32(ip)
+			newIPInt := ipInt + uint32(delta)
+			adjustedIPInt := adjustIPUint32(newIPInt)
```

Comment:
Im seeing the algo now and I dont think you need to do this. The `+` op above already manages overflow at the CPU level atleast for adding (adding `+1` delta into a maxed out `ipInt` will give us 0 automatically). The one Im not sure about is subtraction (what happens in Golang when you force-convert a `-1` `int` type into `uint32` - is this going to give us the max uint32 representation?

Please do check that and lemme know!

Reply:
Yes! Because in two's complement, -1 is represented as a sequence of all binary 1s. When you convert this to uint32, the bits are interpreted as a positive value, resulting in the maximum value that can be represented by uint32.

Reply:
Now do I have to remove the '+' operator? as it is handling both cases

Reply:
So it seems that `+` is handling both the plus and minus cases (overflow or underflow) properly, in which case the `adjustIPUint32` is unnecessary IIUC.

We can remove that and test the logic on 0 & max IPs and see if it consistently works.